codegenerator/cli/templates/dynamic/codegen/src/RegisterHandlers.res.hbs

```diff
@@ -43,7 +43,7 @@ let registerContractHandlers = (
             ],
             events: [
               {{#each contract.events as | event |}}
-              module(Types.{{contract.name.capitalized}}.{{event.name}}),
+              (Types.{{contract.name.capitalized}}.{{event.name}}.register() :> Internal.baseEventConfig),
```

Comment:
My preference is to compose the types. Ie have a field called baseEventConfig rather than coerce